cli/lsp/config.rs

```diff
@@ -236,149 +242,62 @@ pub struct Settings {
 #[derive(Debug)]
 pub struct Config {
   pub client_capabilities: ClientCapabilities,
-  settings: Arc<RwLock<Settings>>,
-  tx: mpsc::Sender<ConfigRequest>,
+  settings: Settings,
```

Comment:
The `RwLock` on this is removed along with the separate thread to call the client (it's now just fired off in a task and returns to use the same `LanguageServer` mutex)

Reply:
Ah, yeah, ok... That whole tx/sender bit was to work around deadlocking issues I remember now, so removing it all together is a big 👍 .